source/common/res/features/split-transaction-auto-adjust/main.js

```diff
@@ -0,0 +1,109 @@
+(function poll() {
+  // Waits until an external function gives us the all clear that we can run (at /shared/main.js)
+  if (typeof ynabToolKit !== 'undefined' && ynabToolKit.pageReady === true) {
+    ynabToolKit.splitTransactionAutoAdjust = (function () {
+      var addAnotherSplit;
+      var splitTransactionRow;
+      var isInitialized = false;
+      var addingAnotherSplit = false;
+      var deletingSplit = false;
+
+      function initialize() {
+        splitTransactionRow = $('.ynab-grid-add-rows');
+        addAnotherSplit = $('.ynab-grid-split-add-sub-transaction');
+
+        splitTransactionRow.on('keyup', '.currency-input .ember-text-field', onKeyPress);
+        splitTransactionRow.on('click', '.ynab-grid-sub-remove', onDeleteSplit);
+        addAnotherSplit.on('click', onAddAnotherSplit);
+      }
+
+      function onKeyPress() {
+        var element = this;
+        var currentInputClass = getCurrentInputClass();
+
+        if ($(element).parents(currentInputClass).length) {
+          autoFillNextRow(this);
+        }
+      }
+
+      function getCurrentInputClass() {
+        var firstRow = $('.ynab-grid-body-row', splitTransactionRow).first();
+        var outflowValue = parseFloat($('.ynab-grid-cell-outflow .ember-text-field', firstRow).val());
+        var inflowValue = parseFloat($('.ynab-grid-cell-inflow .ember-text-field', firstRow).val());
+        return outflowValue > 0 ? '.ynab-grid-cell-outflow' :
+               inflowValue > 0 ? '.ynab-grid-cell-inflow' : false;
+      }
+
+      function onAddAnotherSplit() {
+        addingAnotherSplit = true;
+      }
+
+      function onDeleteSplit() {
+        deletingSplit = true;
+      }
+
+      function autoFillNextRow(currentInputElement) {
+        var inputClass = getCurrentInputClass();
+        var total = parseFloat($(inputClass + ' .ember-text-field', splitTransactionRow.children().eq(0)).val());
```

Comment:
I haven't tested, but this may have trouble in other locales that use 3.567,00 for example. If we get any bug reports about this, using YNAB shared lib's `unformat()` method might be a better way to go than `parseFloat()`.